apps/stats/views.py

```diff
@@ -699,3 +705,73 @@ def render_json(request, addon, stats):
     fudge_headers(response, stats)
     json.dump(stats, response, cls=DecimalJSONEncoder)
     return response
+
+
+class ArchiveMixin(object):
+    """Provides common helper methods for all archive views."""
+    def get_addon(self, request, addon_id):
+        """Fetches an addon by `id` or `slug`"""
+        qset = Addon.with_unlisted.all()
+
+        if not addon_id.isdigit():
+            addon = get_object_or_404(qset, slug=addon_id)
+        else:
+            addon = get_object_or_404(qset, guid=addon_id)
+
+        # If the addon is unlisted it needs either an owner/viewer/dev/support,
+        # or an unlisted addon reviewer.
```

Comment:
Unlisted add-ons don't have stats at all, so we don't need this check, just return a 404 in any case.